src/App.tsx

```diff
@@ -15,48 +18,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort(
+    (prevGood, nextGood) => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return prevGood.length - nextGood.length;
+        case SortType.ALPHABET:
+          return prevGood.localeCompare(nextGood);
+        default:
+          return 0;
+      }
+    },
+  );
+
+  return !isReversed
+    ? visibleGoods
+    : visibleGoods.reverse();
+}
+
+export const App = () => {
+  const [isRev, setIsRev] = useState(false);
+  const [sortBy, setSortBy] = useState<SortType>(SortType.NONE);
+  const [goodsList, setGoodsList] = useState<string[]>();
```

Comment:
<img width="308" alt="image" src="https://user-images.githubusercontent.com/86183007/227976122-c5133afc-2d02-4711-8207-feedeec33660.png">


https://github.com/mate-academy/react_list-of-goods/blob/master/src/App.tsx#L44

Reply:
payed attention